app/workers/service_discovery/refresh_service_worker.rb

```diff
@@ -4,11 +4,15 @@ module ServiceDiscovery
   class RefreshServiceWorker
     include Sidekiq::Worker
 
-    def perform(service_id)
-      return unless ThreeScale.config.service_discovery.enabled
+    def perform(service_id, user_id=nil)
+      user = User.where(id: user_id).first
+      oauth_manager = ServiceDiscovery::OAuthManager.new(user)
+
+      # TODO: add more error reporting here
+      return unless oauth_manager.service_usable?
```

Comment:
This is the same as https://github.com/3scale/porta/blob/0bfb675b807a3c0c14d75da1f719a7be10589e1c/app/workers/service_discovery/create_service_worker.rb#L8-L12

Perhaps we could extract it to a module.